src/silo/query_engine/query_plan.cpp

```diff
@@ -57,20 +66,17 @@ arrow::Status QueryPlan::executeAndWriteImpl(
    };
    SPDLOG_TRACE("Finished reading all batches.");
    auto future = arrow_plan->finished();
-   if (future.state() == arrow::FutureState::PENDING) {
-      SPDLOG_DEBUG(
-         "Plan still pending, but no more results in the future? Not a problem in itself as it "
-         "could be a race condition so the next await will clear this, but be a hint in case we "
-         "detect some weird behavior."
-      );
-   }
-   future.Wait();
```

Comment:
This wait could be a problem, if there's a deadlock it'll hang forever. An await doesn't fix a deadlock.